deepchem/models/trainer.py

```diff
@@ -0,0 +1,64 @@
+import deepchem as dc
+from deepchem.models.lightning.dc_lightning_module import DCLightningModule
+from deepchem.models.lightning.dc_lightning_dataset_module import DCLightningDatasetModule, collate_dataset_wrapper
+
+class DistributedTrainer():
```

Comment:
Let's add more info here about the algorithm, how it works under the hood. Make at least a page of docs. Add to rst

Reply:
Added docs and to rst